src/apps/watchfaces/OswAppWatchface.cpp

```diff
@@ -25,33 +27,34 @@ const char* OswAppWatchface::getAppName() {
 }
 
 #ifdef OSW_FEATURE_STATS_STEPS
-void OswAppWatchface::drawStepHistory(OswUI* ui, uint8_t x, uint8_t y, uint8_t w, uint8_t h, uint32_t max) {
+void OswAppWatchface::drawStepHistory(OswUI* ui, uint8_t posX, uint8_t posY, uint8_t width, uint8_t height, uint32_t max) {
     OswHal* hal = OswHal::getInstance();
     OswUI::getInstance()->resetTextColors();
     uint32_t weekDay = 0;
     uint32_t dayOfMonth = 0;
+    uint8_t index = 0;
     hal->getLocalDate(&dayOfMonth, &weekDay);
-    for (uint8_t i = 0; i < 7; i++) {
-        uint32_t s = hal->environment()->getStepsOnDay(i);
-        uint16_t boxHeight = ((float)(s > max ? max : s) / max) * h;
-        boxHeight = boxHeight < 2 ? 0 : boxHeight;
+    for (index = 0; index < WEEK; index++) {
+        uint32_t stepsCountOnDay = hal->environment()->getStepsOnDay(index);
+        uint16_t boxHeight = ((float)(stepsCountOnDay > max ? max : stepsCountOnDay) / max) * height;
+        boxHeight = boxHeight < 2 ? 0 : boxHeight; // minumum value is 2
 
         // step bars
-        uint16_t c = (unsigned int) OswConfigAllKeys::stepsPerDay.get() <= s ? ui->getSuccessColor() : ui->getPrimaryColor();
-        hal->gfx()->fillFrame(x + i * w, y + (h - boxHeight), w, boxHeight, c);
+        uint16_t stepsCountBoxColor = (unsigned int) OswConfigAllKeys::stepsPerDay.get() <= stepsCountOnDay ? ui->getSuccessColor() : ui->getPrimaryColor();
+        hal->gfx()->fillFrame(posX + index * width, posY + (height - boxHeight), width, boxHeight, stepsCountBoxColor);
         // bar frames
-        uint16_t f = weekDay == i ? ui->getForegroundColor() : ui->getForegroundDimmedColor();
-        hal->gfx()->drawRFrame(x + i * w, y, w, h, 2, f);
+        uint16_t frameLineColor = weekDay == index ? ui->getForegroundColor() : ui->getForegroundDimmedColor();
+        hal->gfx()->drawRFrame(posX + index * width, posY, width, height, 2, frameLineColor); // 2 is cornerRoundness
 
         // labels
         hal->gfx()->setTextCenterAligned();  // horiz.
         hal->gfx()->setTextBottomAligned();
         hal->gfx()->setTextSize(1);
-        hal->gfx()->setTextCursor(CENTER_X, y - 1);
+        hal->gfx()->setTextCursor(CENTER_X, posY - 1);
 
         hal->gfx()->print(hal->environment()->getStepsToday() + (OswConfigAllKeys::settingDisplayStepsGoal.get() ? String("/") + max:""));
 
-        hal->gfx()->setTextCursor(CENTER_X, y + 1 + 8 + w * 4);
+        hal->gfx()->setTextCursor(CENTER_X, 221);
```

Comment:
I would not hard-code such value! The display size is dynamic and the placement of the boxes is too. If you want you could try to tell the compiler to pre-calculate it using a `constexpr`, but this math is really not a slowdown.